zebra-network/src/peer_set/candidate_set/tests/prop.rs

```diff
@@ -17,7 +17,60 @@ proptest! {
         let validated_peers = validate_addrs(gossiped_peers, last_seen_limit);
 
         for peer in validated_peers {
-            prop_assert![peer.get_last_seen() <= last_seen_limit];
+            // Check that malicious peers can't control Zebra's future connections
+            //
+            // Compare timestamps, allowing an extra second, to account for `chrono` leap seconds:
+            // See https://docs.rs/chrono/0.4.19/chrono/naive/struct.NaiveTime.html#leap-second-handling
+            prop_assert!(peer.get_last_seen().timestamp() <= last_seen_limit.timestamp() + 1,
```

Comment:
I'm not sure I understand this change. Why should the timestamps be compared instead of the time itself?

Reply:
Sorry for the confusion here.

I made a follow-up comment about this change, but it seems like it got lost? (Maybe it was internet trouble.)

The `chrono` library stores dates to nanosecond precision. During a leap second, those nanoseconds go up to 2 million. We need to account for these extra nanoseconds in our calculations - in particular, we don't want Zebra to panic during a leap second. (Delaying things by a second should mostly be fine.)

So I made the random `MetaAddr` and `datetime_full` proptest times include nanoseconds in #2195.

But I actually don't think adjusting the tests is very robust. So I normalised the time instead in bc90054. (We only get nanoseconds from OS times. The serialized formats are all in seconds.)